### PR TITLE
my first commit

### PR DESCRIPTION
**Pull request by inna-honcharenko**

### CODE REVIEW

**Review by kudinov-fedor — CHANGES_REQUESTED** [review]
name of module should stick to python naming convention of variables,

only letters numbers and underscore are allowed

File: tests/innahoncharenko/homework_1/test_check_password.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+def test_check_invalid_length(password):
+    assert not check_password(password)
+
+
+@pytest.mark.parametrize("password", [
+    "Aa+11111",
+    "Aa+21111Aa+21111Aa+21111Aa+211",
+])
+def test_check_valid_length(password):
+    assert check_password(password)
```
Comment (fkudinov-mn) [inline tests/innahoncharenko/homework_1/test_check_password.py:62]: duplicate

File: tests/innahoncharenko/homework_1/test_check_password.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+def test_missing_uppercase():
+    assert not check_password("aaaaaa1!")
+
+
+def test_missing_lowercase():
+    assert not check_password("AAAAAAA1!")
+
+
+def test_missing_symbol():
+    assert not check_password("AAAAAAA1")
```
Comment (fkudinov-mn) [inline tests/innahoncharenko/homework_1/test_check_password.py:78]: can use parametrize for  65 - 78 tests

File: tests/innahoncharenko/homework_1/test_check_password.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    assert check_password(f"1aaaaaa!{uppercase_letter}")
+
+@pytest.mark.parametrize("lowercase_letter", "abcdefghijklmnopqrstuvwxyz")
+def test_check_all_lower_letters(lowercase_letter):
+    assert check_password(f"1AAAAAA!{lowercase_letter}")
+
+
+@pytest.mark.parametrize("letter", " !\"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~")
+def test_check_all_special_chars(letter):
+    assert check_password(f"1AAAAAAa{letter}")
```
Comment (fkudinov-mn) [inline tests/innahoncharenko/homework_1/test_check_password.py:97]: there shoould be empty line at end of file

File: tests/innahoncharenko/homework_1/test_not_reliable_function.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    with pytest.raises(RuntimeError):
+        not_reliable()
+
+
+def test_equal_zero(mocker):
+    mocker.patch.object(random, "random", return_value=0)
+    with pytest.raises(RuntimeError):
+        not_reliable()
+
+
```
Comment (fkudinov-mn) [inline tests/innahoncharenko/homework_1/test_not_reliable_function.py:38]: 1 and only 1 empty line at end of file

File: tests/innahoncharenko/homework_1/test_not_reliable_function.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+def test_less_zero_five(mocker):
+    mocker.patch.object(random, "random", return_value=0.4)
+    with pytest.raises(RuntimeError):
+        not_reliable()
+
+
+def test_equal_zero(mocker):
+    mocker.patch.object(random, "random", return_value=0)
+    with pytest.raises(RuntimeError):
+        not_reliable()
```
Comment (fkudinov-mn) [inline tests/innahoncharenko/homework_1/test_not_reliable_function.py:36]: can use parametrize for tests 27 - 33

File: tests/innahoncharenko/homework_1/test_not_reliable_function_2.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+def not_reliable():
+    res = random()
+    if res < 0.5:
+        raise RuntimeError("Res is less than 0.5:  {}".format(res))
+    return res
+
+
+def test_equal_zero_five(mocker):
+    mocker.patch(__name__ + ".random", return_value=0.5)
```
Comment (fkudinov-mn) [inline tests/innahoncharenko/homework_1/test_not_reliable_function_2.py:16]: nice

File: tests/innahoncharenko/homework_1/test_not_reliable_function_2.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    mocker.patch(__name__ + ".random", return_value=0.4)
+    with pytest.raises(RuntimeError):
+        not_reliable()
+
+
+def test_equal_zero(mocker):
+    mocker.patch(__name__ + ".random", return_value=0)
+    with pytest.raises(RuntimeError):
+        not_reliable()
+
```
Comment (fkudinov-mn) [inline tests/innahoncharenko/homework_1/test_not_reliable_function_2.py:35]: 1 and only 1 empty line at end of file

File: tests/innahoncharenko/homework_1/test_use_config.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+def test_config_debug_is_false(monkeypatch):
+    monkeypatch.setitem(CONFIG,'debug', False)
+    assert if_debug_on() is False
+
+
+def test_config_debug_not_present(monkeypatch):
+    monkeypatch.delitem(CONFIG,'debug')
+    with pytest.raises(KeyError):
+        if_debug_on()
```
Comment (fkudinov-mn) [inline tests/innahoncharenko/homework_1/test_use_config.py:34]: missing empty line at end of file

**Review by kudinov-fedor — APPROVED** [review]

**Pull request merged**